icm456xx/imu/inv_imu_transport.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2020 TDK Invensense
+ *
+ * SPDX-License-Identifier: BSD 3-Clause
+ */
+
+/** @defgroup Transport Transport
+ *  @brief    Abstraction layer to communicate with device.
+ *  @{
+ */
+
+/** @file inv_imu_transport.h */
+
+#ifndef _INV_IMU_TRANSPORT_H_
+#define _INV_IMU_TRANSPORT_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+/** @brief Function pointer to read register(s).
+ *  @param[in] reg   Register address to be read.
+ *  @param[out] buf  Output data from the register.
+ *  @param[in] len   Number of byte to be read.
+ *  @return          0 on success, negative value on error.
+ */
+typedef int (*inv_imu_read_reg_t)(void * context, uint8_t reg, uint8_t *buf, uint32_t len);
+
+/** @brief Function pointer to write register(s).
+ *  @param[in] reg  Register address to be written.
+ *  @param[in] buf  Input data to write.
+ *  @param[in] len  Number of byte to be written.
+ *  @return         0 on success, negative value on error.
+ */
+typedef int (*inv_imu_write_reg_t)(void * context, uint8_t reg, const uint8_t *buf, uint32_t len);

```

Comment:
Same question I think I've left in another PR: how will this work with async transfers? (e.g: Streaming mode). Is there async variant that provides the result through a callback?

Reply:
Hi @ubieda , I'm still stuck to merge this. Could you approve it? As said above, the reviewers should be tdk folks but it is zephyr members who are not really available I supposed ... I don't know how to move forward.